design/httpproxy-status-meta.md

```diff
@@ -0,0 +1,52 @@
+# HTTPProxy Status Metadata
+
+Status: Draft
+
+## Abstract
+Since the addition of `Includes` for HTTPProxy, it can be difficult for owners of child HTTPProxy resources to understand what has been included to them without having access to a parent HTTPProxy.
+ 
+Users should have a way to understand this extra metadata about the ingress resources. 
+In addition, with this information, additional tooling can be developed against HTTPProxy resources to drive dashboards (i.e. Octant) and other `kubectl plugins`.
+
+## Goals
+- Expose extra metadata information to HTTPProxy resources
+
+## Detailed Design
+Extend the HTTPProxy.Status struct to include additional fields to support exposing metadata information:
+
+```
+type HTTPProxyStatus struct {
+  ...
+  <existing fields>
+  ...
+  
+  Metadata HTTPProxyMetadata `json:"metadata,omitempty"`
+}
+
+type HTTPProxyMetadata struct {
+  // ParentsRef shows one level parent of any HTTPProxy resources 
+  // which have included this resource.
+  Includes []HTTPProxyParent `json:"includes,omitempty"`
```

Comment:
Why only one level here? Are you expecting that tooling would walk the chain of references to build up the inclusion tree? If the user can't access all the parents, would that limit the ability of the user to see the full context?

Reply:
I was thinking giving users the full chain wouldn't be good since you're exposing the entire cluster. Plus I think it could get out of hand depending on how many includes there might be. Currently, Contour only reasons about having a single include to a child, so my thought was to mirror that and have the child 1. know where they get their includes from 2. Understand the full path of includes that have been passed.

Reply:
Your comment there Steve is a great example of explanation that should be in the high-level design section, outlining what the goals for your design are and how you expect to achieve them.

Reply:
I think that the structs are fine in general, they meet the goal you've put there. I worry that the UX of actually tracing things might take a lot of steps, but I agree that exposing the full include chain is probably too much information.